hexrd/ui/azimuthal_overlay_manager.py

```diff
@@ -0,0 +1,251 @@
+        for mat in materials:
+            cb.addItem(mat, mat)
+
+        cb.setCurrentIndex(materials.index(v))
+        cb.currentIndexChanged.connect(self.update_config_materials)
+        self.material_combos.append(cb)
+        return self.create_table_widget(cb)
+
+    def create_visibility_checkbox(self, v):
+        cb = QCheckBox(self.ui.table)
+        cb.setChecked(v)
+        cb.toggled.connect(self.update_visibilities)
+        self.visibility_boxes.append(cb)
+        return self.create_table_widget(cb)
+
+    def create_table_widget(self, w):
+        # These are required to center the widget...
+        tw = QWidget(self.ui.table)
+        layout = QHBoxLayout(tw)
+        layout.addWidget(w)
+        layout.setAlignment(Qt.AlignCenter)
+        layout.setContentsMargins(0, 0, 0, 0)
+        return tw
+
+    def clear_table(self):
+        self.material_combos.clear()
+        self.visibility_boxes.clear()
+        self.ui.table.clearContents()
+
+    def update_table(self):
+        block_list = [
+            self.ui.table,
+            self.ui.table.selectionModel()
+        ]
+
+        with block_signals(*block_list):
+            prev_selected = self.selected_row
+
+            self.clear_table()
+            self.ui.table.setRowCount(len(self.overlays))
+            for i, overlay in enumerate(self.overlays):
+                w = QTableWidgetItem(overlay['name'])
+                self.ui.table.setItem(i, COLUMNS['name'], w)
+
+                w = self.create_materials_combo(overlay['material'])
+                self.ui.table.setCellWidget(i, COLUMNS['material'], w)
+
+                w = self.create_visibility_checkbox(overlay['visible'])
+                self.ui.table.setCellWidget(i, COLUMNS['visible'], w)
+
+            if prev_selected is not None:
+                select_row = (prev_selected if prev_selected < len(self.overlays)
+                              else len(self.overlays) - 1)
+                self.select_row(select_row)
+
+            self.ui.table.resizeColumnsToContents()
+
+            # The last section isn't always stretching automatically, even
+            # though we have setStretchLastSection(True) set.
+            # Force it to stretch manually.
+            last_column = max(v for v in COLUMNS.values())
+            self.ui.table.horizontalHeader().setSectionResizeMode(
+                last_column, QHeaderView.Stretch)
+
+            # Just in case the selection actually changed...
+            self.selection_changed()
+
+    def select_row(self, i):
+        if i is None or i >= self.ui.table.rowCount():
+            # Out of range. Don't do anything.
+            return
+
+        # Select the row
+        selection_model = self.ui.table.selectionModel()
+        selection_model.clearSelection()
+
+        model_index = selection_model.model().index(i, 0)
+        command = QItemSelectionModel.Select | QItemSelectionModel.Rows
+        selection_model.select(model_index, command)
```

Comment:
It might be good for us to make a common base class between this class and the overlay manager class that contains functions like these where the logic is identical. I might make a quick attempt at this.

Reply:
This makes sense to me but I did not address this in this PR